File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
        int l = 0;
        int r = n - 1;
        while(l < r){
            int m = (l + r) / 2;
            if(nums[m] > nums[r]){
                // Nằm phía bên trái
                l = m + 1;
            }
            else if(nums[m] < nums[r]){
                r = m;
            }
        }
        return nums[l];
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
import bisect as bs
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)

        if n < 1:
            return 0
        
        v = []

        for x in nums:
            # Lower bound
            pos = bs.bisect_left(v, x)

            if pos == len(v):
                v.append(x)
            else:
                v[pos] = x

        return len(v)